addons/base/views.py

```diff
@@ -281,100 +280,205 @@ def get_auth(auth, **kwargs):
         sentry.log_message(str(err))
         raise HTTPError(http_status.HTTP_403_FORBIDDEN)
 
-    if not auth.user:
-        auth.user = OSFUser.from_cookie(data.get('cookie', ''))
 
+def get_authenticated_resource(resource_id):
+    resource = AbstractNode.load(resource_id) or Preprint.load(resource_id)
+    if not resource:
+        raise HTTPError(http_status.HTTP_404_NOT_FOUND, message='Resource not found.')
+
+    """Convert a DraftNode to its corresponding node if applicable."""
+    resource = resource.registered_draft.first() if isinstance(resource, DraftNode) else resource
+
+    if resource.deleted:
+        raise HTTPError(http_status.HTTP_410_GONE, message='Resource has been deleted.')
+
+    return resource
+
+
+def get_file_version_from_wb(waterbutler_data):
+    path = waterbutler_data.get('path')
+    if not path:
+        return
+    file_id = path.strip('/')
+    try:
+        filenode = OsfStorageFileNode.objects.get(_id=file_id)
+    except OsfStorageFileNode.DoesNotExist:
```

Comment:
this is going to raise for any non-osfstorage filenode, and I don't see anywhere prior to this where we're confirming that the provider is `osfstorage`

Reply:
The `path` checking above filters out non-osfstorage providers, but yeah I see what you mean.